binding/binding.go

```diff
@@ -31,12 +41,28 @@ func NewBinding(queue string, exchange string, routingKey string, arguments *amq
 	if topic {
 		var err error
 		if binding.regexp, err = buildRegexp(routingKey); err != nil {
-			// TODO handle that error
-			panic(err)
+			return nil, amqp.NewConnectionError(amqp.CommandInvalid,
+				fmt.Sprintf("queue.bind: %s", err.Error()),
+				amqp.ClassQueue, amqp.MethodQueueBind)
+		}
+	}
+
+	if arguments == nil {
+		return binding, nil
+	}
+
+	argMap := map[string]interface{}(*arguments)
```

Comment:
I thought this had to be done in order to access the attributes of the Table, my bad, if this works without it, I'll remove the statement

Reply:
Since there's no access function on amqp.Table, I left the cast, as otherwise the project would not compile.
If you have a way to make this work without the cast, I'd be happy to remove it though